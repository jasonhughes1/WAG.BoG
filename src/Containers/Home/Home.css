.dog-cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: auto; }

.card-container {
  margin-top: 100px; }
