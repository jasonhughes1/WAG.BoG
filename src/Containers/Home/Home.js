import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import RootReducers from '../../Reducers/RootReducers';
import { connect } from 'react-redux';
import { fetchDogSuccess, fetchDogs} from '../../Actions/Actions';
import Card from '../../Components/Card/Card';
import helpers from '../../Helpers/helpers';
import css from './Home.css';


class Home extends Component {
  render() {
    if(!this.props.dogs.length) {
      return(
        <div>wait</div>
        )
    } else {
      console.log(this.props.dogs[0]);
      const dogCards = this.props.dogs[0].map((dog, index) => {
        console.log(dog.photo);
        return <Card
           name={dog.name}
          breed={dog.breed}
            sex={dog.sex}
        picture={dog.photo}


        />
      })
      return(
        <div className='card-container'>
          <div>{dogCards}</div>
        </div>
        )

      }
    }
  }

const mapStateToProps = (store) => {
  return {
    dogs: store.dogs
  }
}

export default connect(mapStateToProps, null)(Home);
