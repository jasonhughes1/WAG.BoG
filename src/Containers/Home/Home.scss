.dog-cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: auto;
}

.golden-gif {
  height: 100vh;
  width: 100vw;
}

.home-image {
  background-size: cover;
  height: 500px;
  width: 600px;
  align-self: center;
}

.card-container {
  background-color:black;
  display: flex;
  flex-direction: column;
}
