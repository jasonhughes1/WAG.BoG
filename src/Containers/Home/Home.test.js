import React from 'react';
import { shallow } from 'enzyme';
import { Home } from './Home';

describe('Home tests', () => {
  let home;
  beforeEach( () => {
    const mockFunc = jest.fn();
    home = shallow(
      <Home
        dispatch={mockFunc}
        dogs={['pups']}
        history={{key: 'value'}}
        location={{key: 'value'}}
        match={{key: 'value'}}
      />);
  });


  it.skip('should match the snapshot', () => {
    expect(home).toMatchSnapshot();
  });

  it.skip('Should be defined', () => {
    expect(home).toBeDefined();
  });
});
