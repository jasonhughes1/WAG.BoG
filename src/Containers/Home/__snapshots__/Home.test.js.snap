// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    dispatch={[Function]}
    dogs={
        Array [
            Array [
              "pups",
            ],
          ]
    }
    history={
        Object {
            "key": "value",
          }
    }
    location={
        Object {
            "key": "value",
          }
    }
    match={
        Object {
            "key": "value",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="home-image"
          src="ken-reid-290760_Fotor.jpg"
/>,
        <div
          className="dog-cards"
>
          <Connect(Card)
                    breed={undefined}
                    city={undefined}
                    name={undefined}
                    picture={undefined}
                    sex={undefined}
                    state={undefined}
                    street={undefined}
                    zip={undefined}
          />
</div>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "home-image",
          "src": "ken-reid-290760_Fotor.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Card)
              breed={undefined}
              city={undefined}
              name={undefined}
              picture={undefined}
              sex={undefined}
              state={undefined}
              street={undefined}
              zip={undefined}
/>,
          ],
          "className": "dog-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "breed": undefined,
              "city": undefined,
              "name": undefined,
              "picture": undefined,
              "sex": undefined,
              "state": undefined,
              "street": undefined,
              "zip": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="home-image"
            src="ken-reid-290760_Fotor.jpg"
/>,
          <div
            className="dog-cards"
>
            <Connect(Card)
                        breed={undefined}
                        city={undefined}
                        name={undefined}
                        picture={undefined}
                        sex={undefined}
                        state={undefined}
                        street={undefined}
                        zip={undefined}
            />
</div>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "home-image",
            "src": "ken-reid-290760_Fotor.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Card)
                breed={undefined}
                city={undefined}
                name={undefined}
                picture={undefined}
                sex={undefined}
                state={undefined}
                street={undefined}
                zip={undefined}
/>,
            ],
            "className": "dog-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "breed": undefined,
                "city": undefined,
                "name": undefined,
                "picture": undefined,
                "sex": undefined,
                "state": undefined,
                "street": undefined,
                "zip": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
